src/test/regress/expected/isolation_merge.out

```diff
@@ -0,0 +1,147 @@
+Parsed test spec with 2 sessions
+
+starting permutation: s1-begin s1-upd-ins s2-result s1-commit s2-result
```

Comment:
one interesting test would be:

s1-begin s1-upd-ins s2-begin s2-modify-preps s1-commit s2-commit s2-result

where modifying `preps` should also be blocked for citus.shard_replication_factor = 2. For that to happen, task->modifyWithSubquery = true; should be set, but as far as I remember, it is already the case.



Other than that, I guess we probably do not need all the combinations of merge actions, but also does not hurt. So, feel free drop drop some

Reply:
Added a new spec with, but this is result in a deadlock?
```
SET citus.shard_replication_factor TO 2;
permutation "s1-begin" "s1-upd-ins" "s2-begin" "s2-upd-del" "s1-commit" "s2-commit" "s2-result"
```

Reply:
It's all good now, to save time, I was running `make check-isolation-custom-schedule` with only merge tests in it, but we need initial tests (`isolation_setup`) for proper setup. After I include them, there are no issues (the hang is from `pg_isolation_test_session_is_blocked`(), we need to replace it with `citus_isolation_test_session_is_blocked`())

![image](https://user-images.githubusercontent.com/44680808/228046752-c01ae01a-7fe5-412a-82bd-dc3ac2799a92.png)